### PR TITLE
fix menu default reset, add player name to the menu

### PR DESCRIPTION
**Pull request by rfomin**

### CODE REVIEW

File: Source/m_menu.h [hunk 1]
```diff
@@ -117,17 +117,18 @@ extern int warning_about_changes, print_warning_about_changes;
 #define S_DISABLE 0x1000000 // Disable item
 #define S_COSMETIC 0x2000000 // Don't warn about change
 #define S_THERMO  0x4000000 // Mini-thermo
+#define S_NAME    0x8000000 // Player name
```
Comment (fabiangreffrath) [inline Source/m_menu.h:120]: Good idea, better than reusing `S_CHAT`. Can we get rid of `S_FILE`?

**Comment by fabiangreffrath** [comment]
Is item reset behavior now consistent across all menus?

File: Source/m_menu.h (hunk 1 above)
Comment (rfomin) [reply Source/m_menu.h:120]: > Can we get rid of `S_FILE`?

Yes, and we can remove WAD/DEH loader settings from m_misc.c

**Comment by rfomin** [comment]
> Is item reset behavior now consistent across all menus?

Yes, based on my testing.

File: Source/m_menu.h (hunk 1 above)
Comment (fabiangreffrath) [reply Source/m_menu.h:120]: Let's do it.

**Pull request merged**